ddtrace/internal/ci_visibility/writer.py

```diff
@@ -52,9 +59,11 @@ def __init__(
         if not headers["dd-api-key"]:
             raise ValueError("Required environment variable DD_API_KEY not defined")
 
+        self._state = self.STATE_AGENTLESS
+
         super(CIVisibilityWriter, self).__init__(
             intake_url=intake_url,
-            endpoint="api/v2/citestcycle",
+            endpoint=AGENTLESS_ENDPOINT,
```

Comment:
If we set this as the default endpoint, what happens when the `_check_agent()` determines that the agent supports EVP proxy? Does the `CIVisibilityWriter` then change this endpoint back to the agent? Or is this endpoint the same for both agent and agentless?

Reply:
In that situation, the endpoint gets changed [here](https://github.com/DataDog/dd-trace-py/pull/5497/files#diff-8ed1a099c4aaa47a54c8cb4529f67c3462e29a36b171c4fa508605cc644c4282R110).